src/shared/reducers/challenge-listing/index.js

```diff
@@ -270,6 +271,14 @@ export function factory(req) {
 
   if (req) {
     state.filter = req.query.filter;
+    if (!!state.filter && !!state.filter.startDate
+        && moment(state.filter.startDate).isValid() === false) {
```

Comment:
Hmm... It is enough to write
`if (state.filter && state.filter.stateDate && moment(state.filter.startDate).isValid()) {`
not sure, why you don't do it this way? Please fix.